controllers/bookController.js

```diff
@@ -1,114 +1,84 @@
-var Book = require("../models/book");
-var Author = require("../models/author");
-var Genre = require("../models/genre");
-var BookInstance = require("../models/bookinstance");
+const Book = require("../models/book");
+const Author = require("../models/author");
+const Genre = require("../models/genre");
+const BookInstance = require("../models/bookinstance");
 
 const { body, validationResult } = require("express-validator");
+const asyncHandler = require("express-async-handler");
 
-var async = require("async");
-
-exports.index = function (req, res) {
-  async.parallel(
-    {
-      book_count: function (callback) {
-        Book.countDocuments({}, callback);
-      },
-      book_instance_count: function (callback) {
-        BookInstance.countDocuments({}, callback);
-      },
-      book_instance_available_count: function (callback) {
-        BookInstance.countDocuments({ status: "Available" }, callback);
-      },
-      author_count: function (callback) {
-        Author.countDocuments({}, callback);
-      },
-      genre_count: function (callback) {
-        Genre.countDocuments({}, callback);
-      },
-    },
-    function (err, results) {
-      res.render("index", {
-        title: "Local Library Home",
-        error: err,
-        data: results,
-      });
-    }
-  );
-};
+exports.index = asyncHandler(async (req, res, next) => {
+  // Get details of books, book instances, authors and genre counts (in parallel)
+  const [
+    numBooks,
+    numBookInstances,
+    numAvailableBookInstances,
+    numAuthors,
+    numGenres,
+  ] = await Promise.all([
+    Book.countDocuments({}),
+    BookInstance.countDocuments({}),
+    BookInstance.countDocuments({ status: "Available" }),
+    Author.countDocuments({}),
+    Author.countDocuments({}),
+    //Book.find({ author: req.params.id }, "title summary").exec(),
+  ]);
```

Comment:
Remove spurious line
```suggestion
    Author.countDocuments({}),
  ]);
```